### ISSUE TITLE
array([[nan]]) in decoder.cv_scores_

### ISSUE THREAD

**Issue by vukovicnikola** [nilearn/nilearn#1619]
Hi! I was trying to re-fit a SpaceNetClassifier using manually specified alphas from a previous full run. Everything seems to finish without errors, except when I later inspect the decoder object, the cv_scores_ array is "array([[nan]])"?
What could be the problem?
I've tried specifying the alphas parameter both as a single float, as well as a number-of-folds sized array.

**Comment by bthirion** [comment]
Can you provide a code snippet for us to reproduce this issue ?
Best,

**Comment by vukovicnikola** [comment]
Sure!
```
cv = StratifiedKFold(n_folds=10, y=y)
decoder = SpaceNetClassifier(
    cv=cv,
    memory="nilearn_cache",
    penalty='tv-l1',
    alphas = 0.19334502715501783,
    n_alphas = None,
    memory_level=2,
    n_jobs=1,
    verbose=1,
    screening_percentile=20,
    mask=mnimask) # just a regular mni brain mask

decoder.fit(X, y)  # fit
```

**Comment by GaelVaroquaux** [comment]
How did you choose the alpha value that you are using? I am a bit worried
about a manually-set alpha.

**Comment by vukovicnikola** [comment]
It is the mean (best) alpha value from a previous full run of the same script (in which alphas was not prespecified). It's mostly a sanity check for me as an additional analysis.

**Comment by GaelVaroquaux** [comment]
> It is the mean (best) alpha value from a previous full run of the same
> script (in which alphas was not prespecified). It's mostly a sanity
> check for me as an additional analysis.

That's probably the problem, though. That alpha value is probably too
high, and results in a fully sparse model.

**Comment by vukovicnikola** [comment]
But it's the mean of the previous SpaceNet model, with the same data and everything else... Confused.

**Comment by vukovicnikola** [comment]
Hi, @GaelVaroquaux and @bthirion! Any updates perhaps on this issue?

**Comment by KamalakerDadi** [comment]
I am going to look into this today.

**Comment by GaelVaroquaux** [comment]
> But it's the mean of the previous SpaceNet model, with the same data and
> everything else... Confused.

Yes, it's because when you give only one alpha we don't do
cross-validation (line 798 of space_net.py). So it's expected.

If you really want to reproduce things, you need to use the same exact
code.

We could decide to remove this feature, and always do cross-validation.
But I suspect that it would surprise other people.

I am leaning on to concluding that it's not a bug and closing this issue.

**Comment by KamalakerDadi** [comment]
>Yes, it's because when you give only one alpha we don't do cross-validation (line 798 of space_net.py). So it's expected.

Yes. If alphas are specified, Spacenet does not take into account the cross-validation. Because, it assumes best alpha is already provided. It will just fit on the data provided with this alpha. As a result you will see coefficient image fitted with this alpha but not cv_scores_.
Just a pinch of salt.

As @GaelVaroquaux suggested, I think doing a cross-validation is a good idea.

If you really want to Spacenet to take cross-validation specified with alphas, then you need specify more than 1 float values in both [alphas and l1_ratios]. May be it should work with cv_scores_.

>I am leaning on to concluding that it's not a bug and closing this issue.

+1 if people agreed.
